src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsIconFactory.java

```diff
@@ -905,17 +914,30 @@ public void paintIcon(Component c, Graphics g, int x, int y) {
                         if (xp != null) {
                             Skin skin;
                             skin =  xp.getSkin(c, backgroundPart);
-                            skin.paintSkin(g, x, y,
-                                getIconWidth(), getIconHeight(), backgroundState);
-                            if (icon == null) {
+                            if (!isWindows11OrLater) {
+                                skin.paintSkin(g, x, y,
+                                        getIconWidth(), getIconHeight(), backgroundState);
+                                if (icon == null) {
+                                    skin = xp.getSkin(c, part);
+                                    skin.paintSkin(g, x + OFFSET, y + OFFSET, state);
+                                }
+                            } else {
+                                skin.paintSkin(g, x, y,
+                                        getIconWidth(), getIconHeight(), backgroundState);
+                                skinWidth = getIconWidth();
                                 skin = xp.getSkin(c, part);
-                                skin.paintSkin(g, x + OFFSET, y + OFFSET, state);
+                                skin.paintSkin(g, x, y + OFFSET, state);
                             }
                         }
                     }
                 }
                 if (icon != null) {
-                    icon.paintIcon(c, g, x + OFFSET, y + OFFSET);
+                    if (!isWindows11OrLater) {
+                        icon.paintIcon(c, g, x + OFFSET, y + OFFSET);
+                    } else {
+                        icon.paintIcon(c, g, x + icon.getIconWidth(),
+                                y + OFFSET);
+                    }
```

Comment:
If an icon is present, it gets painted instead of check mark or bullet in Windows 10.

In Windows 11 the icon gets painted on

* on the same location in regular case when the current popup menu contains no check or radio menu items with an icon;
* on its own column with has proper large margins when there's at least one check or radio menu item with an icon; the size of the entire popup has to increase to allocate space for the icon column.